nuxeo-core/nuxeo-core-api/src/test/java/org/nuxeo/ecm/core/api/local/DummyLoginFeature.java

```diff
@@ -0,0 +1,105 @@
+/*
+ * (C) Copyright 2019 Nuxeo (http://nuxeo.com/) and others.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * Contributors:
+ *     Kevin Leturc <kleturc@nuxeo.com>
+ */
+package org.nuxeo.ecm.core.api.local;
+
+import javax.security.auth.login.LoginContext;
+import javax.security.auth.login.LoginException;
+
+import org.junit.runners.model.FrameworkMethod;
+import org.nuxeo.ecm.core.api.NuxeoException;
+import org.nuxeo.ecm.core.api.NuxeoPrincipal;
+import org.nuxeo.runtime.api.Framework;
+import org.nuxeo.runtime.api.login.LoginComponent;
+import org.nuxeo.runtime.test.runner.Deploy;
+import org.nuxeo.runtime.test.runner.Features;
+import org.nuxeo.runtime.test.runner.FeaturesRunner;
+import org.nuxeo.runtime.test.runner.RunnerFeature;
+import org.nuxeo.runtime.test.runner.RuntimeFeature;
+
+/**
+ * Feature allowing to mock the {@link LoginComponent} in order to test services which needs security context.
+ * <p>
+ * This feature contributes {@link DummyLoginAs} to the component and registers 3 domains:
+ * <ul>
+ * <li>nuxeo-system-login</li>
+ * <li>nuxeo-client-login</li>
+ * <li>nuxeo-ecm-web</li>
+ * </ul>
+ * By default the feature will log in as Administrator, like the {@link org.nuxeo.ecm.core.test.CoreFeature} does.
```

Comment:
Maybe add that a custom user can be used in a test class or method by adding the `WithUser` annotation?